exercises/list-ops/src/test/java/ListOpsTest.java

```diff
@@ -0,0 +1,364 @@
+        final List<Integer> expected
+                = IntStream.range(0, 4)
+                        .boxed()
+                        .collect(Collectors.toList());
+
+        assertNotNull(actual);
+        assertFalse(actual.isEmpty());
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    @Ignore
+    public void shouldConcatenateTwoListsWithSameElements() {
+        final List<Integer> list1 = Collections.unmodifiableList(
+                IntStream.range(0, 4).boxed().collect(Collectors.toList())
+        );
+        final List<Integer> list2 = Collections.unmodifiableList(
+                IntStream.range(1, 6).boxed().collect(Collectors.toList())
+        );
+        final List<Integer> expected
+                = Stream.concat(list1.stream(), list2.stream())
+                        .collect(Collectors.toList());
+        final List<Integer> actual = ListOps.concat(list1, list2);
+
+        assertNotNull(actual);
+        assertFalse(actual.isEmpty());
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    @Ignore
+    public void shouldConcatenateSeveralLists() {
+        final List<Integer> list1 = Collections.unmodifiableList(
+                IntStream.range(0, 4).boxed().collect(Collectors.toList())
+        );
+        final List<Integer> list2 = Collections.unmodifiableList(
+                IntStream.range(4, 8).boxed().collect(Collectors.toList())
+        );
+        final List<Integer> list3 = Collections.unmodifiableList(
+                IntStream.range(8, 12).boxed().collect(Collectors.toList())
+        );
+        final List<Integer> list4 = Collections.unmodifiableList(
+                IntStream.range(12, 16).boxed().collect(Collectors.toList())
+        );
+        final List<Integer> expected
+                = new ArrayList<>(list1);
+        expected.addAll(list2);
+        expected.addAll(list3);
+        expected.addAll(list4);
+
+        final List<Integer> actual
+                = ListOps.concat(list1, list2, EMPTY_LIST, list3, list4);
+
+        assertNotNull(actual);
+        assertFalse(actual.isEmpty());
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    @Ignore
+    public void shouldReturnIdentityWhenAnEmptyListIsReduced() {
+        final int expected = 0;
+        final int actual
+                = ListOps.reduce(EMPTY_LIST, 0, (x, y) -> x + y, Integer::sum);
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    @Ignore
+    public void shouldCalculateTheSumOfANonEmptyIntegerList() {
+        final List<Integer> list = Collections.unmodifiableList(
+                IntStream.range(0, 5)
+                        .boxed()
+                        .collect(Collectors.toList())
+        );
+        final int actual = ListOps.reduce(list, 0,
+                (x, y) -> x + y,
+                Integer::sum);
+
+        assertEquals(10, actual);
+    }
+
+    @Test
+    @Ignore
+    public void shouldReduceWithAnticommutativeAccumulator() {
```

Comment:
🤔  wow... okay, I'm conflicted.

On the one hand, I love the notion of really taking it deep in this exercise.  This test does that — it's an invitation to really dig into the Java Streams API.

On the other hand, it's *really* deep for someone just getting started.  This might mean we're throwing a bone to those who are already familiar with the API and haven't dug deeply into this more complete form of `reduce`.  Most of what I see in the wild sticks with `collect()`, but that's missing out.  It might also be too big of a jump in this one exercise...

👥  @exercism/java: can I get another opinion on this?